api/base/filters.py

```diff
@@ -20,53 +30,184 @@ def filter_queryset(self, request, queryset, view):
+        """
+        if op not in set(self.MATCH_OPERATORS + self.COMPARISON_OPERATORS + (self.DEFAULT_OPERATOR, )):
+            valid_operators = self._get_valid_operators(field)
+            raise InvalidFilterOperator(value=op, valid_operators=valid_operators)
+        if op in self.COMPARISON_OPERATORS:
+            if not self._isinstance_any(field, self.COMPARABLE_FIELDS):
+                raise InvalidFilterComparisonType(
+                    parameter="filter",
+                    detail="Field '{0}' does not suport comparison operators in a filter.".format(field_name)
+                )
+        if op in self.MATCH_OPERATORS:
+            if not self._isinstance_any(field, self.MATCHABLE_FIELDS):
+                raise InvalidFilterMatchType(
+                    parameter="filter",
+                    detail="Field '{0}' does not support match operators in a filter.".format(field_name)
+                )
+
+    def _parse_date_param(self, field, field_name, op, value):
+        """
+        Allow for ambiguous date filters. This supports operations like findings Nodes created on a given day
+        even though Node.date_created is a specific datetime.
+
+        :return list<dict>: list of one (specific datetime) or more (date range) parsed query params
+        """
+        time_match = self.DATETIME_PATTERN.match(value)
+        if op != 'eq' or time_match:
+            return [{
+                'op': op,
+                'value': self.convert_value(value, field)
+            }]
+        else:  # TODO: let times be as generic as possible (i.e. whole month, whole year)
+            start = self.convert_value(value, field)
+            stop = start + datetime.timedelta(days=1)
+            return [{
+                'op': 'gte',
+                'value': start
+            }, {
+                'op': 'lt',
+                'value': stop
+            }]
+
+    def parse_query_params(self, query_params):
+        """Maps query params to a dict useable for filtering
+        :param dict query_params:
+        :return dict: of the format {
+            <resolved_field_name>: {
+                'op': <comparison_operator>,
+                'value': <resolved_value>
+            }
+        }
+        """
+        query = {}
+        for key, value in query_params.iteritems():
+            match = self.QUERY_PATTERN.match(key)
+            if match:
+                field_name = match.groupdict()['field'].strip()
+                field = self._get_field_or_error(field_name)
+
+                op = match.groupdict().get('op') or self._get_default_operator(field)
+                self._validate_operator(field, field_name, op)
+
+                field_name = self.convert_key(field_name, field)
+                if field_name not in query:
+                    query[field_name] = []
+
+                # Special case date(time)s to allow for ambiguous date matches
+                if type(field) in self.DATE_FIELDS:
```

Comment:
Do we intentionally want this to exclude subclasses of something in DATE_FIELDS?

If not, consider `isinstance(field, self.DATE_FIELDS). It will be a bit more robust for customized date-based fields.